llvm/include/llvm/CodeGen/TargetLowering.h

```diff
@@ -1154,6 +1154,8 @@ class TargetLoweringBase {
     MaybeAlign align = Align(1);   // alignment
 
     MachineMemOperand::Flags flags = MachineMemOperand::MONone;
+
+    AtomicOrdering ordering = AtomicOrdering::NotAtomic;
```

Comment:
This piecemeal addition of fields to `IntrinsicInfo` which ends up not actually capturing the whole breadth of MMO options is problematic.

Could we simply add a `MachineMemOperand *` field in here which, if `nullptr`, supersedes all the other fields?

Reply:
I started refactoring this to add `MachineMemOperand` field, but half way through it seemed to me that the code was more complex due to that change. It would require to construct MMO in multiple places in `getTgtMemIntrinsic` which seemed unnecessary to me. I think we now have all the necessary fields to construct MMO. What do you think?